src/main/java/rs/teslaris/core/service/impl/DocumentPublicationServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+import rs.teslaris.core.service.JournalService;
+import rs.teslaris.core.service.MultilingualContentService;
+import rs.teslaris.core.service.PersonContributionService;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class DocumentPublicationServiceImpl implements DocumentPublicationService {
+
+    private final DocumentRepository documentRepository;
+
+    private final DocumentFileService documentFileService;
+
+    private final MultilingualContentService multilingualContentService;
+
+    private final JournalService journalService;
+
+    private final PersonContributionService personContributionService;
+
+    @Value("${document.approved_by_default}")
+    private Boolean documentApprovedByDefault;
+
+
+    @Override
+    public Document findDocumentById(Integer documentId) {
+        return documentRepository.findById(documentId)
+            .orElseThrow(() -> new NotFoundException("Document with given id does not exist."));
+    }
+
+    @Override
+    public JournalPublicationResponseDTO readJournalPublicationById(Integer publicationId) {
+        var publication = (JournalPublication) findDocumentById(publicationId);
+        if (!publication.getApproveStatus().equals(ApproveStatus.APPROVED)) {
+            throw new NotFoundException("Document with given id does not exist.");
+        }
+        return JournalPublicationConverter.toDTO(publication);
+    }
+
+    @Override
+    public JournalPublication createJournalPublication(JournalPublicationDTO publicationDTO) {
+        var publication = new JournalPublication();
+
+        setCommonFields(publication, publicationDTO);
+        setJournalPublicationRelatedFields(publication, publicationDTO);
+
+        publication.setApproveStatus(
+            documentApprovedByDefault ? ApproveStatus.APPROVED : ApproveStatus.REQUESTED);
+
+        return documentRepository.save(publication);
+    }
+
+    @Override
+    public void editJournalPublication(Integer publicationId,
+                                       JournalPublicationDTO publicationDTO) {
+        var publicationToUpdate = (JournalPublication) findDocumentById(publicationId);
+
+        clearCommonFields(publicationToUpdate);
+        publicationToUpdate.getUris().clear();
+
+        setCommonFields(publicationToUpdate, publicationDTO);
+        setJournalPublicationRelatedFields(publicationToUpdate, publicationDTO);
+    }
+
+    @Override
+    public void deleteJournalPublication(Integer journalPublicationId) {
+        var publicationToDelete = (JournalPublication) findDocumentById(journalPublicationId);
+
+        publicationToDelete.getProofs()
+            .forEach(proof -> documentFileService.deleteDocumentFile(proof.getServerFilename()));
+
+        publicationToDelete.getFileItems().forEach(
+            fileItem -> documentFileService.deleteDocumentFile(fileItem.getServerFilename()));
+
+        documentRepository.delete(publicationToDelete);
```

Comment:
What about ES indexes? Will be implemented later?

Reply:
I wanted to discuss them on the next weekly meeting when we will have basic CRUD operations for proceedings publications, and we can maybe make one index for multiple publication types